test/unit/base-connection-pool.test.js

```diff
@@ -76,7 +76,18 @@ test('API', (t) => {
     let connection = pool.addConnection(href);
     t.same(pool.markDead(connection), pool);
     connection = pool.connections.find((c) => c.id === href);
-    t.equal(connection.status, Connection.statuses.ALIVE);
+    t.equal(connection.status, Connection.statuses.DEAD);
+    t.end();
+  });
+
+  t.test('call markDead twice', (t) => {
```

Comment:
Why do you have to call markDead twice? Calling it once won't work?

Reply:
> Why do you have to call markDead twice? Calling it once won't work?

https://github.com/opensearch-project/opensearch-js/pull/430#pullrequestreview-1336043602